pkg/gatewayserver/io/mqtt/mqtt.go

```diff
@@ -273,6 +282,7 @@ func (c *connection) CanWrite(info *auth.Info, topicParts ...string) bool {
 
 func (c *connection) deliver(pkt *packet.PublishPacket) {
 	logger := log.FromContext(c.io.Context()).WithField("topic", pkt.TopicName)
+	defer recoverMQTTFrontend(c.io.Context())
```

Comment:
We're already in a goroutine that's being recovered here I guess

Reply:
This function seems to be a callback to `mystique` and so I added the recovery here as well.

Reply:
Yes, but is it a callback on our goroutine or does Mystique spin up a goroutine?

Reply:
No clue @htdvisser ?

Reply:
See https://github.com/TheThingsIndustries/mystique

Reply:
Mystique doesn't start a lot of goroutines. As far as I remember, just two per connection (read+write). This would mean that the delivery would be done by the goroutine that calls `session.ReadPacket()`.